multilingual-polylang.php

```diff
@@ -2,101 +2,137 @@
-				foreach ( self::$other_languages as $other_language ) {
-					$duplicate_ids[] = $trans_post[ $other_language ];
-				}
-			}
-		}
-
-		return $duplicate_ids;
-	}
-
-	private static function is_in_current_language( $trans_post ) {
-		return $trans_post[ self::$current_language ] != 0;
-	}
+class MultilingualPolylang
+{
+    private static $current_language;
+    private static $other_languages;
+
+    public static function init()
+    {
+        if (self::is_polylang_installed() && ! is_admin()) {
+            self::$current_language = pll_current_language();
+            self::$other_languages = array_diff(
+                pll_languages_list(),
+                array( self::$current_language )
+            );
+
+            /* Don't redirect URLs in 'wrong' language so that we can
+             * have article in one language and interface in another
+             */
+            add_filter('pll_check_canonical_url', '__return_false');
+
+            global $polylang;
+
+            remove_filter('option_sticky_posts', array( $polylang->filters, 'option_sticky_posts' ));
+        }
+    }
+
+    /**
+     * Check if polylang is installed and active
+     *
+     * @version 1.0.0
+     * @return  bool
+     */
+
+    private static function is_polylang_installed()
+    {
+        return function_exists('pll_languages_list');
+    }
+
+    /**
+     * Creates a new query for all posts avabile in any language
+     *
+     * @version 1.0.3
+     * @return  WP_Query
+     */
+
+    public static function get_query($args = array())
+    {
+        /* https://wordpress.org/support/topic/show-posts-from-other-languages/?replies=18
+         * by way of
+         * http://wordpress.syllogic.in/2014/08/going-multi-lingual-with-polylang/
+         */
+        $duplicated_post_ids = self::get_duplicated_posts();
+
+        $all_languages = '';
+
+        $defaults = array(
+            'lang' => $all_languages,
+            'post__not_in' => $duplicated_post_ids,
+        );
+
+        $args = array_merge($defaults, $args);
+
+        return new WP_Query($args);
+    }
+
+    /**
+     * A replacement for the WordPress function get_permalink()
+     *
+     * This will replace the language in a post URL so that a post can be viewed
+     * in a language different to that of the rest of the interface.
+     *
+     * @version 1.0.0
+     * @return  string A url to the current language
+     */
+    public static function get_permalink($post)
+    {
+        $link = get_permalink($post);
+
+        if (self::is_polylang_installed()) {
+            $post_language = pll_get_post_language($post->ID);
+
+            if ($post_language != self::$current_language) {
+                $search = "/$post_language/";
+                $replace = '/' . self::$current_language . '/';
+
+                $link = str_replace($search, $replace, $link);
+            }
+        }
+
+        return (string) $link;
+    }
+
+    /**
+     * Get duplicates of posts
```

Comment:
The method description doesn't really describe what the method does.

We don't want to list a post in more than one language if a translation exists in the current language. So this method gets the IDs of all the duplicate posts to exclude from the query. These are the posts that are translations of the current language but not the post in the current language.

I can't think of a shorter way to describe this. Maybe you can?